### PR TITLE
Can't load libsignal_jni in windows env #484

### PR DESCRIPTION
**Pull request by ClementBareth**
#484

### CODE REVIEW

File: java/shared/java/org/signal/libsignal/internal/Native.java [hunk 1]
```diff
@@ -49,8 +49,14 @@ private static void copyToTempFileAndLoad(InputStream in, String extension) thro
   private static void loadLibrary() {
     try {
       String  osName    = System.getProperty("os.name").toLowerCase(java.util.Locale.ROOT);
-      boolean isMacOs   = osName.startsWith("mac os");
-      String  extension = isMacOs ? ".dylib" : ".so";
+      String extension;
+      if (osName.startsWith("mac os")) {
+        extension = ".dylib";
+      } else if(osName.contains("win")) {
```
Comment (jrose-signal) [inline java/shared/java/org/signal/libsignal/internal/Native.java:55]: Does `startsWith` not work? Seems good to keep the check narrow.

**Review by jrose-signal — COMMENTED** [review]
Thank you for picking this up! This has been broken since, uh, forever, but since libsignal_jni is primarily used for the Signal server and Android app, and none of the server and Android developers are doing their day-to-day work on Windows, we've been letting it slip by.

It's a *little* trickier than what you have here, because the canonical Windows library name we've been shipping is "signal_jni.dll", not "libsignal_jni.dll". But that's just making the check cover the whole filename rather than just the extension.

File: java/shared/java/org/signal/libsignal/internal/Native.java (hunk 1 above)
Comment (ClementBareth) [reply java/shared/java/org/signal/libsignal/internal/Native.java:55]: I think the startsWIth should work indeed

**Comment by jrose-signal** [comment]
(closed due to age and no pressing need from within Signal, anyone should feel free to pick it up and finish it off, along with testing on a Windows machine)

**Pull request closed without merging**